internal/provider/kubernetes/predicates.go

```diff
@@ -61,10 +62,14 @@ type NamespaceGetter interface {
 }
 
 // checkObjectNamespaceLabels checks if labels of namespace of the object is a subset of namespaceLabels
-// TODO: check if param can be an interface, so the caller doesn't need to get the namespace before calling
-// this function.
-func (r *gatewayAPIReconciler) checkObjectNamespaceLabels(nsString string) (bool, error) {
-	// TODO: add validation here because some objects don't have namespace
+func (r *gatewayAPIReconciler) checkObjectNamespaceLabels(obj matav1.Object) (bool, error) {
+
+	var nsString string
+	// it requires extra condition?
```

Comment:
is this comment a TODO?

Reply:
Strictly speaking, this may be a TODO. Because we need to judge cluster resources and some resources that do not have namespace.

Reply:
got it! can you make this comment more clear? it will help those who come across it later.

Reply:
yes, i have do that